### ISSUE TITLE
Require name scope

### ISSUE THREAD

**Issue by piranna** [tclindner/npm-package-json-lint#200]
Add new rule to require name to have defined a scope. Ideally It should be a three state field (always, never or don't café).

**Comment by piranna** [comment]
Maybe valid-name-scope would do the job with a regexp, in that case an example is needed un the docs.

**Comment by tclindner** [comment]
Thank you, @piranna! I take a look at this tomorrow and get back to you.

**Comment by piranna** [comment]
Thank you! :-)

**Comment by tclindner** [comment]
@piranna do you think [`valid-values-name-scope`](https://npmpackagejsonlint.org/docs/en/rules/valid-values/valid-values-name-scope) will meet your needs? `error` covers `always`. `off` covers `don't care`. `never` is a gap.

btw - I've got time this weekend to get caught up on some tickets 🤓

**Comment by piranna** [comment]
> @piranna do you think [`valid-values-name-scope`](https://npmpackagejsonlint.org/docs/en/rules/valid-values/valid-values-name-scope) will meet your needs? `error` covers `always`. `off` covers `don't care`. `never` is a gap.

Didn't test it, but as I've told you in https://github.com/tclindner/npm-package-json-lint/issues/200#issuecomment-578499019, reviewing the code maybe it could work by defining a regular expression instead of a full scope, just only it would need to be tested :-)

By the way, why dows it requires the `@` to be defined in the name scope? Is it for generity in case there's another registry don't use that pattern for scopes?

**Comment by tclindner** [comment]
@piranna - yes, re: `@`. Do you have any concerns with that approach? 

I'm going to focus on adding plugin support next. Do you have an immediate use case for `never` right now? If so, I can focus on this first.

**Comment by piranna** [comment]
> @piranna - yes, re: `@`. Do you have any concerns with that approach?

Not at all, just only would be nice to have that in the documentation :-)


> Do you have an immediate use case for `never` right now?

Not at all :-)
